patient-appointment-management/functions/helpers.private.js

```diff
@@ -0,0 +1,331 @@
+        APPLICATION_CUSTOMER_CODE
+      );
+    }
+    case 'AWS_GLUE_CRAWLER': {
+      const APPLICATION_CUSTOMER_CODE = await retrieveParameter(
+        context,
+        'APPLICATION_CUSTOMER_CODE'
+      );
+      return (
+        CONSTANTS.AWS_GLUE_CRAWLER_BASE_NAME + '-' + APPLICATION_CUSTOMER_CODE
+      );
+    }
+    case 'AWS_LAMBDA_SEND_REMINDERS': {
+      const APPLICATION_CUSTOMER_CODE = await retrieveParameter(
+        context,
+        'APPLICATION_CUSTOMER_CODE'
+      );
+      return (
+        CONSTANTS.AWS_LAMBDA_SEND_REMINDERS_BASE_NAME +
+        '-' +
+        APPLICATION_CUSTOMER_CODE
+      );
+    }
+    case 'AWS_S3_BUCKET': {
+      const APPLICATION_CUSTOMER_CODE = await retrieveParameter(
+        context,
+        'APPLICATION_CUSTOMER_CODE'
+      );
+      return (
+        CONSTANTS.AWS_S3_BUCKET_BASE_NAME + '-' + APPLICATION_CUSTOMER_CODE
+      );
+    }
+    case 'TWILIO_ACCOUNT_SID': {
+      return retrieveParameter(context, 'ACCOUNT_SID');
+    }
+    case 'TWILIO_AUTH_TOKEN': {
+      return retrieveParameter(context, 'AUTH_TOKEN');
+    }
+    case 'TWILIO_ENVIRONMENT_SID': {
+      const service_sid = await retrieveParameter(
+        context,
+        'TWILIO_SERVICE_SID'
+      );
+      if (service_sid === null) {
+        return null; // service not yet deployed
+      }
+      const environments = await client.serverless
+        .services(service_sid)
+        .environments.list();
+      console.debug(environments[0]);
+      return environments[0].sid;
+    }
+    case 'TWILIO_ENVIRONMENT_DOMAIN_NAME': {
+      const service_sid = await retrieveParameter(
+        context,
+        'TWILIO_SERVICE_SID'
+      );
+      if (service_sid === null) {
+        return null; // service not yet deployed
+      }
+      const environments = await client.serverless
+        .services(service_sid)
+        .environments.list();
+      return environments[0].domainName;
+    }
+    case 'TWILIO_FLOW_SID': {
+      let flow_sid = null;
+      await client.studio.flows.list({ limit: 100 }).then((flows) =>
+        flows.forEach((f) => {
+          if (f.friendlyName === CONSTANTS.APPLICATION_NAME) {
+            flow_sid = f.sid;
+          }
+        })
+      );
+      if (flow_sid !== null) {
+        /*
+         * do not save in case flow is manually deleted
+         * await assignParameter(context,'TWILIO_FLOW_SID', flow_sid);
+         */
+        return flow_sid;
+      }
+      return null;
+    }
+    case 'TWILIO_SERVICE_SID': {
+      let service_sid = null;
+      await client.serverless.services.list({ limit: 100 }).then((services) =>
+        services.forEach((s) => {
+          if (s.friendlyName === CONSTANTS.APPLICATION_NAME) {
+            service_sid = s.sid;
+          }
+        })
+      );
+      if (service_sid !== null) {
+        console.log('Found', service_sid);
+        return service_sid;
+      }
+      console.log('YOU MUST DEPLOY THE SERVICE FIRST!!! ABORTING!!!');
```

Comment:
This is a bit shout-y; consider a quick sentence on where the user can go to deploy the service.

Reply:
I'll take this out, as it's not needed.